apps/distgo/params/dist.go

```diff
@@ -136,3 +148,29 @@ type RPMDistInfo struct {
 func (i *RPMDistInfo) Type() DistInfoType {
 	return RPMDistType
 }
+
+func (i *RPMDistInfo) Deps() []string {
+	// no deps for rpm type
+	return []string{}
+}
+
+type DockerDeps map[string][]DistInfoType
+
+type DockerDistInfo struct {
+	Repository string
+	Tag        string
+	ContextDir string
+	DistDeps   DockerDeps
+}
+
+func (d *DockerDistInfo) Type() DistInfoType {
+	return DockerDistType
+}
+
+func (d *DockerDistInfo) Deps() []string {
+	var deps []string
+	for product := range d.DistDeps {
+		deps = append(deps, product)
+	}
+	return deps
```

Comment:
Note that, because this iterates over map contents, the return order here will be random between calls. This may be fine, but if you want this to be deterministic, you can do so by adding a `sort.Strings(deps)` before the return to ensure that the items are always sorted lexicographically.